File: AngularFront/src/app/interfaces/PlaylistDTO.ts
import { PlaylistitemDTO } from "./PlaylistitemDTO";

export interface PlaylistDTO {
  playListId?: number;
  playListName: string;
  playListDescription: string;
  viewCount?: number;
  likeCount?: number;
  addedCount?: number;
  sharedCount?: number;
  playListImage?: string | null;
  showImage?: string | null;
  playListCreatedAt?: Date;
  playListUpdatedAt?: Date;
  analysisTimestamp?: Date;
  showLikeEffect?: boolean;
  isCreatedByUser?: boolean; // 新增此屬性來區分自建或協作
  videos?: PlaylistitemDTO[];
}

File: VdbAPI/Models/PlaylistDTO.cs
namespace VdbAPI.Models
{
    public class PlaylistDTO
    {
        public int PlayListId { get; set; }
        public string PlayListName { get; set; }
        public string PlayListDescription { get; set; }
        public int ViewCount { get; set; }
        public int LikeCount { get; set; }
        public int AddedCount { get; set; }
        public int SharedCount { get; set; }
        public string ShowImage { get; set; }
    }
}
